### DISCUSSION TITLE
SDcard access

### DISCUSSION THREAD

**Discussion by dresco** [grblHAL/STM32F7xx#10] [General]
Hi Terje,

I notice you've updated to the latest ST library with the new Ethernet code. I think it's in my best interests to have a minimal set of changes, so I'll try and bring my H7 port into line with your LWIP directory structure (I built mine from example code, as from memory the CubeMX generated code wasn't quite correct at the time).

Do you have any thoughts on using the SDMMC peripheral for 4-bit SDcard access? I think that's the biggest difference between the two drivers now..

I had a small SDcard breakout assembled for the Nucleo headers & and have spares, happy to just stick one in the post to you if useful?

![image](https://user-images.githubusercontent.com/996823/196395466-285a04fe-b1c4-4b17-9399-656693cd05be.png)

**Comment by terjeio** [comment]
As long as the FatFs version links with the grblHAL VFS wrapper IMO it does not matter which it is. A faster interface is better, at least for those who want to use the WebUI or use the machine for high speed laser engraving?

> I had a small SDcard breakout assembled for the Nucleo headers & and have spares

This fits into the CN8 SDMMC header pins?

>  happy to just stick one in the post to you if useful?

That would be nice - I can then update the driver for 4-bit access...

Terje Io
Ievegen 120
5917 Rossland
Norway

**Reply by dresco** [reply]
> A faster interface is better, at least for those who want to use the WebUI or use the machine for high speed laser engraving?

Yes I was thinking of the performance potential.. (And for the H7, it's the way the WeAct & BTT boards are designed).

> This fits into the CN8 SDMMC header pins?

Yep
![image](https://user-images.githubusercontent.com/996823/196894567-d681fc20-72ef-405c-95df-544e8ec9829a.png)

> I can then update the driver for 4-bit access...

Cool, got your address, will get one into the post..
